main.go

```diff
@@ -0,0 +1,28 @@
+package main
```

Comment:
>> But how I originally thought about these was that they'd be separate processes, and the server would only accept registrations from the privileged places (ie. only registrations from localhost)

>I would like to count @Liamsi in, since we all agree on this. 

I also thought about it in exactly this way. But I also agree on the security concerns @c633 raised below.

> We can check if the registration requests are originated from the localhost or not. (something like conn.RemoteAddr() == conn.LocalAddr()). But I don't know whether there is any type of attacks that can fake the remote address?

I fear this alone doesn't give you any reasonable security. We have to find ways to additionally secure that (firewall/network setup, TLS). But all this makes only sense if we want both processes also to be able to run on different machines (on the same machine they could use other means to communicate with each other).